File: lab-1/PlayerModule/player.rb
module PlayerModule
  
    # a class represents a single player's information
    # @attr_reader player_name [String] name of the player
    # @attr_reader player_score [Number] Score of the player
    # @author Lucas Wu, Jing Wen, Songyuan Wu
      class Player
        attr_reader  :player_name;
        attr_reader  :player_score;
        attr_accessor  :win_prob;
        attr_reader  :is_human;
        # initialize the information of the new player
        #
        # @param name [String] the name of the new player
        # @return [Player] a new instance of type player
        def initialize(name)
          @player_name = name;
          @player_score = 0;
          @is_human=true
        end
        def change_to_computer()
          @is_human=false
        end
        # update the score of the player won or lost
        #
        # @param points [Number] points that the player won or lost
        # @return the updated score
        def score_add!(points)
    
          @player_score += points;
          @player_score = 0 if @player_score < 0;
          @player_score
        end
    
        # update the score of the player won or lost
        #
        # @param name [Number] the players's new score
        # @return the new score
        def score_change_to(score )
          @player_score = score;
          score
        end
        # update the score of the player won or lost
        #
        # @param name [String] the players's new name
        # @return the new name
        def name_change_to(name)
          @player_name = name;
          name
        end



      end
    end

File: lab-2/Model/PlayerModule/Player.js
/**
 * a class represents a single player's information
 * 
 * @member {string} playerName- name of the player
 * @member {number} playerScore - score of the player
 * @member {number} winProb- wining probability of computer player
 * @member {boolean} isHuman - is human player or computer player
 * 
 * @author Songyuan Wu & Jing Wen
 */

class Player {
    /**
     * Initializes a new instance of the new player
     * 
     * @param	{string} name - name of the player
     *
     * @returns {Player} a new instance of a Player Object
     */
    constructor(name) {
        this.playerName = name;
        this.playerScore = 0;
        this.isHuman = true;
    }

    /**
     * change the player to computer player.
     */
    change_to_computer() {
        this.isHuman = false;
    }

    /**
     * update the score of the player won or lost.
     * 
     * @param {number} points - points that the player won or lost.
     * @returns {number} the updated score.
     */
    scoreAdd(points) {
        this.playerScore += points;
        if (this.playerScore < 0) {
            this.playerScore = 0;
        }
        return this.playerScore
    }

    /**
     * update the score of the player won or lost.
     * 
     * @param {number} score - the player's new score.
     * @returns {number} the new score of the player.
     */
    scoreChangeTo(score) {
        this.playerScore = score;
        return this.playerScore;
    }

    /**
     * update the name of the player.
     * 
     * @param {string} name- the player's new name.
     * @returns {string} the new name of the player.
     */
    nameChangeTo(name) {
        this.playerName = name;
        return this.playerName;
    }
}